src/grammar/block.rs

```diff
@@ -63,6 +64,57 @@ pub(super) fn parse_stmt(p: &mut Parser) {
     p.finish();
 }
 
+fn parse_execute_immediate_stmt(p: &mut Parser) {
+    p.start(SyntaxKind::ExecuteImmediateStmt);
+    p.expect(T![execute]);
+    p.expect(T![immediate]);
+    // Parse String
+    if !p.eat(T![unquoted_ident]) {
+        p.expect(T![quoted_literal]);
+    }
+    // handle INTO, USING and RETURN/RETURNING clauses
+    if p.at(T![into]) {
+        parse_into_clause(p, true);
+    }
+    if p.at(T![using]) {
+        // parse using clause
+        parse_using_clause(p);
+    }
+    if [T![return], T![returning]].contains(&p.current()) {
+        // parse return into stuff
+        parse_return_into_clause(p);
+    }
+    p.eat(T![;]);
+    p.finish();
+}
+
+fn parse_using_clause(p: &mut Parser) {
+    p.start(SyntaxKind::UsingClause);
+    p.expect(T![using]);
+    safe_loop!(p, {
+        if [T![in], T![out]].contains(&p.current()) {
+            if p.at(T![in]) {
+                p.eat(T![in]);
+                p.eat(T![out]);
+            } else {
+                p.eat(T![out]);
+            }
+        }
```

Comment:
Couldn't this be simplified to the following?
```suggestion
        p.eat(T![in]);
        p.eat(T![out]);
```